File: templates/api-universal/api_cms.py
"""
SPS-CMS Universal Python Drop-in Router (FastAPI & Flask compatible)
"""
from fastapi import APIRouter, HTTPException
from pydantic import BaseModel
from typing import Dict, Any, Optional
import sqlite3
import json

router = APIRouter(prefix="/api/cms", tags=["SPS-CMS"])

class SavePageRequest(BaseModel):
    slug: str
    content: Dict[str, Any]

def get_db():
    conn = sqlite3.connect("data/sps_cms.db")
    conn.row_factory = sqlite3.Row
    return conn

@router.post("/save-page")
def save_page(req: SavePageRequest):
    conn = get_db()
    cursor = conn.cursor()
    content_json = json.dumps(req.content)
    cursor.execute("""
        INSERT INTO sps_pages (slug, title, content_json)
        VALUES (?, ?, ?)
        ON CONFLICT(slug) DO UPDATE SET content_json=excluded.content_json, updated_at=CURRENT_TIMESTAMP
    """, (req.slug, req.slug, content_json))
    conn.commit()
    conn.close()
    return {"success": True}

@router.get("/collections/{collection_name}")
def get_collection(collection_name: str, only_homepage: bool = False):
    conn = get_db()
    cursor = conn.cursor()
    query = "SELECT * FROM sps_collections WHERE collection_name = ?"
    params = [collection_name]
    if only_homepage:
        query += " AND show_on_homepage = 1"
    query += " ORDER BY order_index ASC, created_at DESC"
    cursor.execute(query, params)
    rows = cursor.fetchall()
    items = []
    for row in rows:
        d = dict(row)
        d["data"] = json.loads(d["data_json"]) if d.get("data_json") else {}
        items.append(d)
    conn.close()
    return {"items": items}

File: templates/api-universal/api-cms.php
<?php
/**
 * SPS-CMS Universal PHP Drop-in API Handler
 * Compatible with Native PHP, cPanel Shared Hosting, Laravel, and Slim.
 */
header('Content-Type: application/json');
header('Access-Control-Allow-Origin: *');
header('Access-Control-Allow-Headers: Content-Type');

$dbHost = getenv('DB_HOST') ?: 'localhost';
$dbName = getenv('DB_NAME') ?: 'sps_cms';
$dbUser = getenv('DB_USER') ?: 'root';
$dbPass = getenv('DB_PASS') ?: '';

try {
    $pdo = new PDO("mysql:host=$dbHost;dbname=$dbName;charset=utf8mb4", $dbUser, $dbPass, [
        PDO::ATTR_ERRMODE => PDO::ERRMODE_EXCEPTION,
        PDO::ATTR_DEFAULT_FETCH_MODE => PDO::FETCH_ASSOC
    ]);
} catch (PDOException $e) {
    echo json_encode(['error' => 'Database connection failed: ' . $e->getMessage()]);
    exit;
}

$action = $_GET['action'] ?? '';
$input = json_decode(file_get_contents('php://input'), true);

if ($action === 'save_page') {
    $slug = $input['slug'] ?? '/';
    $contentJson = json_encode($input['content'] ?? []);
    
    $stmt = $pdo->prepare("INSERT INTO sps_pages (slug, title, content_json) VALUES (?, ?, ?) 
                           ON DUPLICATE KEY UPDATE content_json = VALUES(content_json), updated_at = CURRENT_TIMESTAMP");
    $stmt->execute([$slug, $slug, $contentJson]);
    echo json_encode(['success' => true]);
    exit;
}

if ($action === 'get_collection') {
    $collection = $_GET['collection'] ?? '';
    $stmt = $pdo->prepare("SELECT * FROM sps_collections WHERE collection_name = ? ORDER BY order_index ASC, created_at DESC");
    $stmt->execute([$collection]);
    $items = $stmt->fetchAll();
    foreach ($items as &$item) {
        $item['data'] = json_decode($item['data_json'], true);
    }
    echo json_encode(['items' => $items]);
    exit;
}

echo json_encode(['error' => 'Invalid action']);
